core/app/models/spree/order.rb

```diff
@@ -400,7 +400,7 @@ def deliver_order_confirmation_email
 
     # Helper methods for checkout steps
     def paid?
-      payment_state == 'paid' || payment_state == 'credit_owed'
+      %w(paid credit_owed).include?(payment_state)
```

Comment:
I'll bring up the concept of immutable data structures with the core team at our meeting next week.

@GeekOnCoffee We're currently talking about some improvements to the coding style & patterns we use in Solidus so we can set a pattern of higher quality for the future. It's a discussion that's just started, and are encouraging suggestions from the community on.